libsolidity/analysis/PostTypeChecker.cpp

```diff
@@ -421,6 +434,111 @@ struct ReservedErrorSelector: public PostTypeChecker::Checker
 	}
 };
 
+class YulLValueChecker : public solidity::yul::ASTWalker
+{
+public:
+	YulLValueChecker(ASTString const& _identifierName): m_identifierName(_identifierName) {}
+	bool willBeWrittenTo() { return m_willBeWrittenTo; }
+	using solidity::yul::ASTWalker::operator();
+	void operator()(solidity::yul::Assignment const& _assignment) override
+	{
+		for (auto const& yulIdentifier: _assignment.variableNames)
+			if (yulIdentifier.name.str() == m_identifierName)
+				m_willBeWrittenTo = true;
```

Comment:
How many `yulIdentifier`s can there be here? Wouldn't it make more sense to return as soon as we assign to `m_willBeWrittenTo`?

```suggestion
			if (yulIdentifier.name.str() == m_identifierName)
			{
				m_willBeWrittenTo = true;
				return;
			}
```

Reply:
Yeah, makes sense, no need to check further.